### ISSUE TITLE
Queries for column names (tag keys) with predicates containing columns that don't exist aren't quite right

### ISSUE THREAD

**Issue by carols10cents** [influxdata/influxdb_iox#4786]
# What happens

Given a table that contains columns `region` and `time`, (and say all region values are "west"), if you do an `InfluxRPC` query for column names with the predicate `region = "west" AND host != "server01"`, you'll get the empty set for the column names.

In other words, this test currently passes:

```
#[tokio::test]
async fn list_tag_columns_predicate_negative_nonexistent_column() {
    let predicate = PredicateBuilder::default()
        .add_expr(col("state").eq(lit("MA"))) // state=MA
        .add_expr(col("host").not_eq(lit("server01"))) // nonexistent column with !=; always true
        .build();
    let predicate = InfluxRpcPredicate::new(None, predicate);
    let expected_tag_keys = vec!["city", "county", "state"];
    // This currently returns nothing, which is incorrect, it should return "city", "county",
    // "state" because a nonexistent column is always not equal to anything.
    let expected_tag_keys = vec![];
    run_tag_keys_test_case(TwoMeasurementsManyNulls {}, predicate, expected_tag_keys).await;
}
```

# What I expected to happen

Because the table doesn't contain the `host` column at all, then `host != "server01"` should always evaluate to `true`. The `region = "west"` expression should be applied, and because in this example all region values in the table are "west", then the result should be all columns, or `[region, time]`.

In other words, this test should pass instead:

```
#[tokio::test]
async fn list_tag_columns_predicate_negative_nonexistent_column() {
    let predicate = PredicateBuilder::default()
        .add_expr(col("state").eq(lit("MA"))) // state=MA
        .add_expr(col("host").not_eq(lit("server01"))) // nonexistent column with !=; always true
        .build();
    let predicate = InfluxRpcPredicate::new(None, predicate);
    let expected_tag_keys = vec!["city", "county", "state"];
    run_tag_keys_test_case(TwoMeasurementsManyNulls {}, predicate, expected_tag_keys).await;
}
```

# More context

- https://github.com/influxdata/influxdb_iox/pull/4767
- https://github.com/influxdata/influxdb_iox/pull/4776
